src/apps/migrate/answers/run.py

```diff
@@ -77,8 +78,17 @@ async def migrate(self):
                                 session=regular_session,
                                 response=mongo_answer,
                             )
-                            respondent_id = mongoid_to_uuid(
-                                mongo_answer["meta"]["subject"].get("@id")
+                            respondent_mongo_id = Profile().findOne(
+                                {
+                                    "_id": mongo_answer["meta"]["subject"].get(
+                                        "@id"
+                                    )
+                                }
+                            )["userId"]
+                            respondent_id = (
+                                mongoid_to_uuid(respondent_mongo_id)
+                                if respondent_mongo_id
+                                else uuid.uuid4()
```

Comment:
why don't we call here get_anonymous_respondent?

Reply:
if you see further, it goes to get_anonymous_respondent function, once finds out that respondent_id doesn't exist